contracts/message-based/PriceAwareUpgradeable.sol

```diff
@@ -107,32 +119,42 @@ contract PriceAwareUpgradeable is OwnableUpgradeable {
 
         require(block.timestamp - dataTimestamp < maxDelay, "Data is too old");
 
+        //We iterate directly through call data to extract the values of symbols
 
-        //Debugging logs (to be removed)
-
-        //    console.log("Len: ", messageLength);
-        //    console.logBytes(rawData);
-        //    console.logBytes32(hash);
-        //    console.logBytes(signature);
-        //    console.log("Signer: ", signer);
-
+        return _readFromCallData(symbols, uint256(dataSize), messageLength);
+    }
 
-        //8. We iterate directly through call data to extract the value for a given symbol
 
-        uint256 val;
-        uint256 max = dataSize;
-        bytes32 currentSymbol;
+    function _readFromCallData(bytes32[] memory symbols, uint256 dataSize, uint16 messageLength) private view returns (uint32[] memory) {
+        bytes32[] memory resultSymbols;
+        uint32[] memory values;
         uint256 i;
+        uint256 j;
+        uint256 readyAssets;
+        bytes32 currentSymbol;
+
         assembly {
             let start := sub(calldatasize(), add(messageLength, 66))
-            for {i := 0} lt(i, max) {i := add(i, 1)} {
-                val := calldataload(add(start, add(32, mul(i, 64))))
+
+            values := msize()
+            mstore(add(values, 0), mload(symbols))
```

Comment:
What's the purpose of add(values,0) ?

Reply:
it doesn't make much sense, replacing with just `values`